datastream-server-restli/src/main/java/com/linkedin/datastream/server/dms/DatastreamResources.java

```diff
@@ -761,6 +761,14 @@ public CreateResponse create(Datastream datastream) {
         metadataMap.put(DatastreamMetadataConstants.IS_USER_MANAGED_DESTINATION_KEY, "true");
       }
 
+      String datastreamName = datastream.getName();
+      String trimmedDatastreamName = (datastreamName == null) ? datastreamName : datastreamName.trim();
```

Comment:
can  datastreamName be null?

Reply:
Removed the null check. Turns out this is already covered by this call performed at the beginning of this method:
```
DatastreamUtils.validateNewDatastream(datastream);
```